src/image.rs

```diff
@@ -465,62 +487,96 @@ pub struct ImageBlit {
+    pub next_access: Access,
+    pub old_layout: Option<SyncLayout>,
+    pub new_layout: SyncLayout,
     pub range: SubresourceRange,
 }
 
 impl<'a> LayoutTransition<'a> {
     pub fn transition_whole(
         image: &'a Image,
-        access: Range<AccessFlags>,
-        layout: Range<Layout>,
+        access: Range<Access>,
+        layout: Range<SyncLayout>,
     ) -> Self {
         LayoutTransition {
             range: SubresourceRange::whole(image.info()),
             image,
-            old_access: access.start,
-            new_access: access.end,
+            prev_access: access.start,
+            next_access: access.end,
             old_layout: Some(layout.start),
             new_layout: layout.end,
         }
     }
 
-    pub fn initialize_whole(image: &'a Image, access: AccessFlags, layout: Layout) -> Self {
+    pub fn initialize_whole(image: &'a Image, next_access: Access, layout: SyncLayout) -> Self {
         LayoutTransition {
             range: SubresourceRange::whole(image.info()),
             image,
-            old_access: AccessFlags::empty(),
+            prev_access: Access::None,
+            next_access,
             old_layout: None,
-            new_access: access,
             new_layout: layout,
         }
     }
 }
 
-#[derive(Clone, Copy, Debug, Hash, PartialEq, Eq)]
+/// Image barriers should only be used when a queue family ownership transfer
+/// or an image layout transition is required - prefer global barriers at all
+/// other times.
+///
+/// In general it is better to use image barriers with `Layout::Optimal`
+/// than it is to use global barriers with images using the
+/// `RawLayout::General` layout.
+///
+/// Access types are defined in the same way as for a global memory barrier, but
+/// they only affect the image subresource range identified by `image` and
+/// `range`, rather than all resources.
+///
+/// An image barrier defining a queue ownership transfer needs to be executed
+/// twice - once by a queue in the source queue family, and then once again by a
+/// queue in the destination queue family, with a semaphore guaranteeing
+/// execution order between them.
+///
+/// If `old_layout` is empty, the contents of the image become
+/// undefined after the barrier is executed, which can result in a performance
+/// boost over attempting to preserve the contents. This is particularly useful
+/// for transient images where the contents are going to be immediately overwritten.
+/// A good example of when to use this is when an application re-uses a presented
+/// image after acquiring the next swap chain image.
+#[derive(Clone, Debug, Hash, PartialEq, Eq)]
 pub struct ImageMemoryBarrier<'a> {
     pub image: &'a Image,
-    pub old_access: AccessFlags,
-    pub old_layout: Option<Layout>,
-    pub new_access: AccessFlags,
-    pub new_layout: Layout,
+    pub prev_access: Access,
+    pub next_access: Access,
+    pub old_layout: Option<SyncLayout>,
+    pub new_layout: SyncLayout,
```

Comment:
Instead of using `SyncLayout` we could offer calculation of optimal `Layout` in methods that build this type.